aeron-client/src/main/c/util/aeron_symbol_table.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2014-2021 Real Logic Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if defined(__linux__)
+#define _BSD_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <errno.h>
+
+#include "util/aeron_dlopen.h"
+#include "util/aeron_error.h"
+#include "aeron_symbol_table.h"
+
+
+static void* aeron_symbol_table_obj_scan(const aeron_symbol_table_obj_t *table, size_t table_size, const char *symbol)
+{
+    void* result = NULL;
+
+    for (size_t i = 0; i < table_size; i++)
+    {
+        const char *alias = table[(int)i].alias;
+        const char *name = table[(int)i].name;
+
+        if (NULL == alias || NULL == name)
+        {
+            break;
+        }
+
+        if (0 == strcmp(alias, symbol) || 0 == strcmp(name, symbol))
```

Comment:
should consider looking at length instead of a naked `strcmp` that may not terminate.